### PR TITLE
LibJS: Implement Number.prototype.to[Exponential,Precision]

### PR DESCRIPTION
**Pull request by trflynn89**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/NumberPrototype.cpp [hunk 1]
```diff
@@ -133,6 +152,138 @@ JS_DEFINE_NATIVE_FUNCTION(NumberPrototype::to_locale_string)
...
+    // 2. If precision is undefined, return ! ToString(x).
+    if (precision_value.is_undefined())
+        return js_string(vm, MUST(number_value.to_string(global_object)));
+
+    // 3. Let p be ? ToIntegerOrInfinity(precision).
+    auto precision = TRY(precision_value.to_integer_or_infinity(global_object));
+
+    // 4. If x is not finite, return ! Number::toString(x).
+    if (!number_value.is_finite_number())
+        return js_string(vm, number_value.to_string_without_side_effects());
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/NumberPrototype.cpp:324]: I think I'd prefer a `MUST(number_value.to_string(global_object))` here

File: Userland/Libraries/LibJS/Runtime/NumberPrototype.cpp [hunk 2]
```diff
@@ -89,6 +120,127 @@ static ThrowCompletionOr<Value> this_number_value(GlobalObject& global_object, V
...
+
+    // 2. Let f be ? ToIntegerOrInfinity(fractionDigits).
+    auto fraction_digits = TRY(fraction_digits_value.to_integer_or_infinity(global_object));
+
+    // 3. Assert: If fractionDigits is undefined, then f is 0.
+    VERIFY(!fraction_digits_value.is_undefined() || (fraction_digits == 0));
+
+    // 4. If x is not finite, return ! Number::toString(x).
+    if (!number_value.is_finite_number())
+        return js_string(vm, number_value.to_string_without_side_effects());
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/NumberPrototype.cpp:139]: Ditto

**Review by linusg — COMMENTED** [review]
Very nice! Two small suggestions.

File: Userland/Libraries/LibJS/Runtime/NumberPrototype.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/NumberPrototype.cpp]: Fixed!

**Pull request merged**